### PR TITLE
Remove dead KNNRetriever class

### PR DESCRIPTION
**Pull request by xacrimon**
This code was previously used by assist but is now deprecated and not used anywhere. We had to drop it due to API restrictions palced upon us by the tree library we were using.

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

File: lib/ai/knnretriever.go [hunk 1]
```diff
@@ -1,119 +0,0 @@
...
-	"github.com/kyroy/kdtree"
-
-	"github.com/gravitational/teleport/lib/ai/embedding"
-)
-
-// Document is a embedding enriched with similarity score
-type Document struct {
-	*embedding.Embedding
-	SimilarityScore float64
-}
```
Comment (hugoShaka) [inline lib/ai/knnretriever.go:30]: I think this type is needed

File: lib/ai/knnretriever.go [hunk 2]
```diff
@@ -1,119 +0,0 @@
...
-		return 0, trace.BadParameter("vectors must be the same length")
-	}
-
-	var result float64
-	for i, val := range v1 {
-		result += val * v2[i]
-	}
-
-	return result, nil
-}
```
Comment (hugoShaka) [inline lib/ai/knnretriever.go:119]: This should also be preserved somwhere as it is actually used

**Review by jakule — COMMENTED** [review]
Compiler agrees
```
lib/ai/simpleretriever.go:77:94: undefined: Document
lib/ai/simpleretriever.go:79:21: undefined: Document
lib/ai/simpleretriever.go:92:20: undefined: calculateSimilarity
lib/ai/simpleretriever.go:95:31: undefined: Document
lib/ai/simpleretriever.go:108:31: undefined: Document
```

**Comment by xacrimon** [comment]
@hugoShaka Yep, just noticed when i saw the test results first. That was a quick review! Fixed

**Comment by xacrimon** [comment]
cc @jakule

**Review by jakule — APPROVED** [review]

**Pull request merged**